compiler/modules/types/mapping.bal

```diff
@@ -407,13 +407,18 @@ function bddMappingMemberType(Context cx, Bdd b, string? key, SemType accum) ret
 }
 
 
-function mappingAtomicMemberType(MappingAtomicType atomic, string? key) returns SemType {
+function mappingAtomicMemberType(MappingAtomicType atomic, StringSubtype? key) returns SemType {
     if key != () {
-        int? i = atomic.names.indexOf(key);
-        if i != () {
-            return atomic.types[i];
+        SemType m = NEVER;
+        foreach var i in 0 ..< atomic.names.length() {
+            if stringSubtypeContains(key, atomic.names[i]) {
```

Comment:
This is O(N^2). atomic.names is sorted, so we can do much better.

I would add a function in string.bal that takes a StringSubtype and a sorted array of strings and returns a list of the indices of the strings that are in StringSubtype. Do this in a way that is O(log(N)). As a subroutine have a function that takes two sorted lists of strings and returns the list of the indices in the second array and that are in the first array. You can use enumerableListIntersect as the basis.